PROTOCOL.md

```diff
@@ -2,179 +2,287 @@
-
-* When receiving new SPDY streams which include the header key "*libchan-ref*", the receiver MUST lookup that
-header value in the table of pending byte streams. If the value is registered in the table, that SPDY stream
-MUST be passed to the application.
-
-On either end, once the SPDY stream for a byte-stream value is established, it MUST be exposed to the application
-as follows:
-
-* After sending each of those SPDY streams, each write operation by the application to a byte-stream field MUST
-trigger the sending of a single data frame on the corresponding SPDY stream.
-
-* Each read operation by the application from a byte-stream field MUST yield the content of the next
-data frame received on the corresponding SPDY stream. If the reading end of the SPDY stream is closed,
-the read operation MUST yield EOF.
-
-* A close operation by the application on the a byte-stream field MUST trigger the closing of the writing end
-of the corresponding SPDY stream.
-
-#### Optional method: file descriptor passing
+* When decoding a message including 1 or more byte stream values, the receiver
+MUST store the unique identifier of each value in a session-wide table of
+pending byte streams. It MAY then immediately pass the decoded message to the
+application.
 
-*FIXME*
+* The sender SHOULD cap the size of its pending byte streams table to a
+reasonable value. It MAY make that value configurable by the application. If it
+receives a message with 1 or more byte stream references, and the table
+is full, the sender MAY suspend processing of the message until there is room in
+the table.
 
-### Sending nested channels
+* When receiving new streams which include the header key "*libchan-ref*", the
+receiver MUST lookup that header value in the table of pending byte streams. If
+the value is registered in the table, that stream MUST be passed to the
+application.
 
-*FIXME*
+On either end, once the stream for a byte-stream value is established, it MUST
+be exposed to the application as follows:
+
+* After sending each of those streams, each write operation by the application
+to a byte-stream field MUST trigger the sending of a single data frame on the
+corresponding stream.
+
+* Each read operation by the application from a byte-stream field MUST yield
+the content of the next data frame received on the corresponding stream. If the
+reading end of the stream is closed, the read operation MUST yield EOF.
+
+* A close operation by the application on the a byte-stream field MUST trigger
+the closing of the writing end of the corresponding SPDY stream.
+
+## Message encoding
+A message may be any type which supported by the libchan encoder.  A libchan
+message encoder must support encoding raw byte stream types as well as channels.
+In addition to the libchan data types, time must also be encoded as a custom
+type to increase portability of the protocol.
+
+Currently supported message encoders are msgpack and soon CBOR.
```

Comment:
Let's specify msgpack (version 5), some older encoder do not support all the features we need.